src/StackExchange.Redis/CommandRetry/CommandRetryPolicy.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+
+namespace StackExchange.Redis
+{
+    /// <summary>
+    /// Policy that determines which commands should be retried upon restoration of a lost connection.
+    /// </summary>
+    public abstract class CommandRetryPolicy
+    {
+        /// <summary>
+        /// Creates a policy instance for a specific multiplexer and its commands.
+        /// </summary>
+        /// <param name="muxer">The muleiplexer this policy is for.</param>
+        protected CommandRetryPolicy(ConnectionMultiplexer muxer)
+        {
+        }
+
+        internal abstract bool TryQueue(Message message, Exception ex);
+
+        /// <summary>
+        /// Returns the current length of the retry queue.
+        /// </summary>
+        public abstract int CurrentQueueLength { get; }
```

Comment:
For exception messages, probably want to `CurrentlyProcessing` (bool) for exception messages